### PR TITLE
[Android] Added a scroll bar for collection view

### LINKED ISSUE #11369: CollectionView shows no ScrollBar
### Description

When filling a CollectionView with more data than fits on the screen I can scroll CollectionView, but no ScrollBar is shown. Not even if I set the `*ScrollBarVisibility` to `Always` or just do this with the `VerticalScrollBarVisibility` only.

### Steps to Reproduce

Create a new .NET MAUI .NET7 App in VS2022 and change those to files to the following (or load my reproduction repository):

MainPage.xaml
```xaml
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage
    x:Class="CollectionViewScrollBar.MainPage"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:collectionViewScrollBar="clr-namespace:CollectionViewScrollBar"
    x:DataType="collectionViewScrollBar:MainPage">

    <CollectionView
        HorizontalScrollBarVisibility="Always"
        ItemsSource="{Binding Rows}"
        VerticalScrollBarVisibility="Always">
        <CollectionView.ItemTemplate>
            <DataTemplate x:DataType="collectionViewScrollBar:Row">
                <Grid RowDefinitions="Auto, Auto">
                    <Label
                        Grid.Row="0"
                        FontAttributes="Bold"
                        Text="{Binding Title}" />
                    <Label
                        Grid.Row="1"
                        FontAttributes="Italic"
                        Text="{Binding Detail}" />
                </Grid>
            </DataTemplate>
        </CollectionView.ItemTemplate>
    </CollectionView>

</ContentPage>
```

MainPage.xaml.cs
```csharp
using System.Collections.ObjectModel;

namespace CollectionViewScrollBar;

public partial class MainPage : ContentPage
{
	public MainPage()
	{
		InitializeComponent();
        BindingContext = this;

        for (var i = 0; i < 50; i++)
        {
            Rows.Add(new Row($"Title-{i}", $"Detail-{i}"));
        }
	}

    public ObservableCollection<Row>  Rows { get; } = new ();
}

public record Row(string Title, string Detail);
```
image

With a ListView it works:
MainPage.xaml
```xaml
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage
    x:Class="CollectionViewScrollBar.MainPage"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:collectionViewScrollBar="clr-namespace:CollectionViewScrollBar"
    x:DataType="collectionViewScrollBar:MainPage">

    <ListView ItemsSource="{Binding Rows}">
        <ListView.ItemTemplate>
            <DataTemplate x:DataType="collectionViewScrollBar:Row">
                <TextCell Detail="{Binding Detail}" Text="{Binding Title}" />
            </DataTemplate>
        </ListView.ItemTemplate>
    </ListView>

</ContentPage>
```
image

### Link to public reproduction project repository

https://github.com/SokoFromNZ/MauiCollectionViewScrollBar

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android 11

### Did you find any workaround?

No

### Relevant log output

_No response_

### LINKED ISSUE #25468: [Android] Collection view has no scroll bar
### Description

https://github.com/user-attachments/assets/ecb2fd13-9f5c-4f13-96a6-2714f7314282

### Steps to Reproduce

```
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="Maui.Controls.Sample.MainPage"
             xmlns:local="clr-namespace:Maui.Controls.Sample">
        <CollectionView
            VerticalScrollBarVisibility="Always"
            x:Name="CollectionView"/>
</ContentPage>
```

```c#
public MainPage()
{
    InitializeComponent();
    var items = Enumerable.Range(1, 100).Select(i => i.ToString()).ToList();
    CollectionView.ItemsSource = items;
}
``` 

### Link to public reproduction project repository

_No response_

### Version with bug

9.0.0-rc.2.24503.2

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

_No response_

### Affected platforms

Android

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kubaflo**
### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/25468
Fixes https://github.com/dotnet/maui/issues/11369

|Before|After|
|--|--|
|<video src="https://github.com/user-attachments/assets/8ae8caa1-3f21-4362-b401-348201b95292" width="300px"/>|<video src="https://github.com/user-attachments/assets/98e8af0b-9c3f-42f3-9fd6-93e0415729a0" width="300px"/>|

### CODE REVIEW

File: src/Controls/tests/TestCases.HostApp/Issues/Issue25468.xaml.xaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
```
Comment (kubaflo) [inline src/Controls/tests/TestCases.HostApp/Issues/Issue25468.xaml.xaml:1]: Change this file name from Issue25468.xaml.xaml to Issue25468.xaml after uploading screenshots

File: src/Controls/tests/TestCases.HostApp/Issues/Issue25468.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿namespace Maui.Controls.Sample.Issues
+{
```
Comment (kubaflo) [inline src/Controls/tests/TestCases.HostApp/Issues/Issue25468.cs:2]: Change this file name from Issue25468.cs to Issue25468.xaml.cs after uploading screenshots

**Review by jfversluis — CHANGES_REQUESTED** [review]
Test fails :( 

image

**Comment by kubaflo** [comment]
@jfversluis hmmm it looks like there are the same problems with other tests too. Could you please /azp again?

**Review by jfversluis — APPROVED** [review]

**Pull request merged**